### PR TITLE
Makes it possible for all Officers and all NCO's (E4+) to recommend people for medals.

### PR DESCRIPTION
**Pull request by TheGamerdk**
# About the pull request

I've been meaning to do this.

Corporal and above + all officers can recommend medals.

# Explain why it's good for the game
# Testing Photographs and Procedure
I tested it.

Tried to give a medal recommendation as a PFC: Failed
Tried to give a medal recommendation as a FTL: Success
Tried to give a medal recommendation as an SO: Success
Tried to give a medal recommendation as XO: Success

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
add: All officers and NCO's can now recommend people for medals.
/:cl:

### CODE REVIEW

File: code/game/objects/items/devices/radio/headset.dm [hunk 1]
```diff
@@ -417,8 +417,10 @@
 	var/obj/item/card/id/id_card = wearer.wear_id?.GetID()
 	if(!istype(id_card))
 		return
-	if(!(id_card.rank in list(JOB_SO, JOB_XO, JOB_SQUAD_LEADER)))
-		to_chat(wearer, SPAN_WARNING("Only Staff Officers, Executive Officers and Squad Leaders are permitted to give medal recommendations!"))
+	var/paygrade = get_paygrade_id_by_name(wearer.get_paygrade(FALSE))
+	var/last_number = text2num(copytext(paygrade, length(paygrade), length(paygrade)+1))
+	if(!(findtext(paygrade, "O") || last_number >= 4))
+		to_chat(wearer, SPAN_WARNING("Only officers or NCO's (ME4+) can recommend medals!"))
```
Comment (Drulikar) [inline code/game/objects/items/devices/radio/headset.dm:423]: String manipulation is not the way to go about this. Either just explicitly list out the ranks that are allowed, or consider a more appropriate value e.g. `officer_grade` on the `/datum/paygrade`

**Review by Drulikar — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/radio/headset.dm (hunk 1 above)
Comment (TheGamerdk) [reply code/game/objects/items/devices/radio/headset.dm:423]: You’ll be having a really long list, and officer_grade doesn’t account for NCOs.
There’s nothing wrong with string manipulation.

**Comment by TheGamerdk** [comment]
Not going to spend time coming up with an inferior solution

**Comment by ProfanityTM** [comment]
Does this apply to ASO? They couldn't do medal recomendations last I recall.

**Comment by TheGamerdk** [comment]
> Does this apply to ASO? They couldn't do medal recomendations last I recall.

They’re an officer, so yes :)

File: code/datums/medal_awards.dm [hunk 2]
```diff
@@ -479,6 +479,10 @@ GLOBAL_LIST_INIT(xeno_medals, list(XENO_SLAUGHTER_MEDAL, XENO_RESILIENCE_MEDAL,
 			continue
 		recipient_ranks[recipient_name] = record.fields["rank"]
 		possible_recipients += recipient_name
+	if(length(possible_recipients) == 0)
```
Comment (vero5123) [inline code/datums/medal_awards.dm:482]: if u wanted u could just do if(!length(possible_recipients))

[the remark this answers is not in the document]
Comment (Drulikar) [reply code/datums/medal_awards.dm:482]: Either are fine.

**Review by Drulikar — APPROVED** [review]
Of note, this allows a CO to make medal recommendations, but I guess this would allow you to prepare a medal?

**Pull request merged**